### ISSUE TITLE
feature request: keyboard shortcuts

### ISSUE THREAD

**Issue by nickPreloaded** [playcanvas/editor#408]
Could be great to be able to change the keyboard shortcuts, especially for rotate, scale, move etc. They're a row higher on the keyboard that other 3d apps seem to be by standard and is playing havoc with my muscle memory. thanks

**Comment by willeastcott** [comment]
So the clash here is with the Editor's fly cam which is WASD (With QE used to raise and lower the view along the vertical axis). So QWE were not available for Translate/Rotate/Scale shortcuts. But I totally get that this is quite non-standard compared to some other apps.

Hey @Maksims - what's your take on this?

**Comment by Maksims** [comment]
> So the clash here is with the Editor's fly cam which is WASD (With QE used to raise and lower the view along the vertical axis). So QWE were not available for Translate/Rotate/Scale shortcuts - 1/2/3 were assigned instead. But I totally get that this is quite non-standard compared to some other apps.
> 
> Hey @Maksims - what's your take on this?

Changing it now. Would definitely be very alienating to current user base.
WASD is so common for many users, loosing it. Or hiding behind other key modifier (like right mouse) would reduce its usability.

Current controls is very much generic across games, where 1-9 used for skills/weapons/inventory slots, and WASD for movement.

I would not change it, unless we have a majority of Editor users consensus for such a change.

**Comment by nickPreloaded** [comment]
The difference here i would argue is that we're not 'playing' the game (where you'd expect wasd) but 'making' the game. Maya, zbrush, Unity, Unreal, all use QWER to switch their tools. 

You guys obviously do what you feel best from feedback as a whole - it was just an observation as a first time playcanvas user.

**Comment by willeastcott** [comment]
To avoid confusing existing users, it's perfectly possible to make shortcut configuration and Editor option. For example, make current behavior enabled for all existing users and for new users, disable that with the default as @nickPreloaded describes.

The important question for me is - which is fundamentally the best option, forgetting about the existing learned behavior.

**Comment by nickPreloaded** [comment]
of course. Yes - having the ability to edit them would be a win all around I think :)

**Comment by yaustar** [comment]
For the short term, we can have user preference for 'Unity' and 'PlayCanvas' keymaps, similar to what a lot of IDEs have.

Webstorm for example has plugins and default keymaps to other text editors such as Visual Studio and Visual Code.

We can later add 'Custom' or others in the future.

It wouldn't be unreasonable to accommodate first time users that migrate from Unity given the size of their userbase.

**Comment by nickPreloaded** [comment]
sounds good. I think Blender does something similar.  The mouse/camera toggling options is something that also often gets customised

**Comment by willeastcott** [comment]
I really like the sound of that, @yaustar. Shouldn't be too difficult to do that.

**Comment by yaustar** [comment]
Bumped by client

**Comment by Mr-D-Anderson** [comment]
Bump
